examples/solids/qfunctions/hyperFSCUR.h

```diff
@@ -0,0 +1,684 @@
+                   (*qdata)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[1];
+  // F is used for hyperelasticity (non-linear)
+  const CeedScalar (*gradu)[3][CEED_Q_VLA] = (const CeedScalar(*)[3][CEED_Q_VLA])in[2];
+
+  // Outputs
+  CeedScalar (*deltadvdX)[3][CEED_Q_VLA] = (CeedScalar(*)[3][CEED_Q_VLA])out[0];
+  // *INDENT-ON*
+
+  // Context
+  const Physics context = (Physics)ctx;
+  const CeedScalar E  = context->E;
+  const CeedScalar nu = context->nu;
+
+  // Constants
+  const CeedScalar TwoMu = E / (1 + nu);
+  const CeedScalar mu = TwoMu / 2;
+  const CeedScalar Kbulk = E / (3*(1 - 2*nu)); // Bulk Modulus
+  const CeedScalar lambda = (3*Kbulk - TwoMu) / 3;
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Read spatial derivatives of delta_u
+    // *INDENT-OFF*
+    const CeedScalar deltadu[3][3] = {{deltaug[0][0][i],
+                                       deltaug[1][0][i],
+                                       deltaug[2][0][i]},
+                                      {deltaug[0][1][i],
+                                       deltaug[1][1][i],
+                                       deltaug[2][1][i]},
+                                      {deltaug[0][2][i],
+                                       deltaug[1][2][i],
+                                       deltaug[2][2][i]}
+                                     };
+    // -- Qdata
+    const CeedScalar wdetJ      =      qdata[0][i];
+    const CeedScalar dXdx[3][3] =    {{qdata[1][i],
+                                       qdata[2][i],
+                                       qdata[3][i]},
+                                      {qdata[4][i],
+                                       qdata[5][i],
+                                       qdata[6][i]},
+                                      {qdata[7][i],
+                                       qdata[8][i],
+                                       qdata[9][i]}
+                                      };
+    // *INDENT-ON*
+
+    // Compute graddeltau
+    //   dXdx = (dx/dX)^(-1)
+    // Apply dXdx to deltadu = graddelta
+    CeedScalar graddeltau[3][3];
+    for (CeedInt j = 0; j < 3; j++)     // Component
+      for (CeedInt k = 0; k < 3; k++) { // Derivative
+        graddeltau[j][k] = 0;
+        for (CeedInt m =0 ; m < 3; m++)
+          graddeltau[j][k] += dXdx[m][k] * deltadu[j][m];
+      }
+
+    // I3 : 3x3 Identity matrix
+    // Deformation Gradient : F = I3 + gradu
+    // *INDENT-OFF*
+    const CeedScalar F[3][3] =  {{gradu[0][0][i] + 1,
+                                  gradu[0][1][i],
+                                  gradu[0][2][i]},
+                                 {gradu[1][0][i],
+                                  gradu[1][1][i] + 1,
+                                  gradu[1][2][i]},
+                                 {gradu[2][0][i],
+                                  gradu[2][1][i],
+                                  gradu[2][2][i] + 1}
+                                };
```

Comment:
Jacobian action should not require building F. This algorithm appears to be accessing more data than Algorithm 2 of Davydov. What happened here?

Reply:
I may have missed something, but per our Jan 26 email:
```
The issue here is we don't want to store F when evaluating the action of the Jacobian. We have

x_current = x_initial + u

We also have a constitutive relation \tau(\bm b, J) from Eq 14, and we're discretizing Eq 7 of Davydov, which can be written

\int_{B_current} \sigma : symgrad(du) = \int_{B_initial} J \sigma : symgrad(du)

where \tau = J \sigma and symgrad is in the current configuration. Their notation is clumsy, IMO, since we're really doing the integral in the current configuration so I'd just use \sigma (Cauchy stress). We know that

dF = \partial du/\partial x_initial
   = (\partial du/\partial x_current) (\partial x_current/\partial x_initial)
   = (\partial du/\partial x_current) F
   = grad(du) F

So you need to be able to evaluate db(du), which is

  db = dF F^T + F dF^T 
     = grad(du) F F^T + F F^T grad(du)^T
     = grad(du) b + b grad(du)^T

if I haven't messed something up in typing this out.
```

Reply:
This code right now answer this question: Is it better to implement hyper-elasticity with constitutive equation in terms of S and C, or tau and b?!

Reply:
So I'm proposing that we should store (during residual evaluation or Jacobian "assembly") the derivative of reference with respect to current so that the Jacobian need only access that, not derivative of reference with respect to initial *and* derivative of initial with respect to current. It cuts down memory access and flops, and should give us a speed boost.

Reply:
If I understood correctly, you propose to save
dx_ref/dx_cur
right now dXdx which is dx_ref/dx_initial provided as input for residual and Jacobian. So we need to do
dx_ref/dx_cur = dx_ref/dx_initial * dx_initial/dx_cur = dx_ref/dx_initial * F^{-1}
(I think this is what you wrote in email)

Reply:
Yes, exactly. See also the derivation in Wriggers
![image](https://user-images.githubusercontent.com/3303/110707768-06eb1500-81b7-11eb-91b8-c3236cce74c6.png)
and Davydov
![image](https://user-images.githubusercontent.com/3303/110707811-14a09a80-81b7-11eb-9c0e-0bd4994dbcdb.png)

Reply:
One more question, in Jacobian we have to compute tau, so we need Grad(u) to compute F then left Cauchy tenser b, then tau.
I think now we need to store Grad(u) and also dx_ref/dx_cur (or let say F^{-1}), am I right?
and if we want to store 2 variables in current configuration, why not do the same thing reference configuration by saving Grad(u) and C^{-1}
![Screenshot from 2021-03-10 15-51-30](https://user-images.githubusercontent.com/42816410/110708832-84fbeb80-81b8-11eb-8c00-21a0131883aa.png)